apollo-biz/src/test/java/com/ctrip/framework/apollo/biz/service/ReleaseHistoryServiceTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.biz.service;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import com.ctrip.framework.apollo.biz.AbstractIntegrationTest;
+import com.ctrip.framework.apollo.biz.config.BizConfig;
+import com.ctrip.framework.apollo.biz.entity.Release;
+import com.ctrip.framework.apollo.biz.entity.ReleaseHistory;
+import com.ctrip.framework.apollo.biz.repository.ReleaseHistoryRepository;
+import com.ctrip.framework.apollo.biz.repository.ReleaseRepository;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import java.lang.reflect.Method;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.springframework.aop.framework.AopProxyUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.Sql.ExecutionPhase;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.util.ReflectionUtils;
+
+/**
+ * @author kl (http://kailing.pub)
+ * @since 2023/3/24
+ */
+public class ReleaseHistoryServiceTest extends AbstractIntegrationTest {
```

Comment:
Please add some test cases where transaction rollback occurs in `com.ctrip.framework.apollo.biz.service.ReleaseHistoryService#cleanReleaseHistory` method.

Reply:
This suggestion is great. I encountered problems with transaction rollback in JUnit test cases and spent a lot of time trying to solve it. However, in the end, it was resolved. It is worth noting that both the updated `TransactionTemplate` and the original `PlatformTransactionManager` class are working properly.